core/src/main/java/org/apache/gravitino/utils/PrincipalUtils.java

```diff
@@ -47,10 +47,19 @@ public static <T> T doAs(Principal principal, PrivilegedExceptionAction<T> actio
   public static Principal getCurrentPrincipal() {
     java.security.AccessControlContext context = java.security.AccessController.getContext();
     Subject subject = Subject.getSubject(context);
-    if (subject == null || subject.getPrincipals(UserPrincipal.class).isEmpty()) {
+    if (subject == null) {
       return new UserPrincipal(AuthConstants.ANONYMOUS_USER);
     }
-    return subject.getPrincipals(UserPrincipal.class).iterator().next();
+
+    if (!subject.getPrincipals(UserPrincipal.class).isEmpty()) {
+      return subject.getPrincipals(UserPrincipal.class).iterator().next();
+    }
+
+    if (!subject.getPrincipals().isEmpty()) {
+      return new UserPrincipal(subject.getPrincipals().iterator().next().getName());
```

Comment:
This logic isn't right. Why do we need this?

Reply:
I think we'd better add some description that `PrincipalUtils` can't only be used in nested `doAs` code block if without this change.